### PR TITLE
Fix non determinism in Zinc analysis store files

### PR DESCRIPTION
**Pull request by jjudd**

### CODE REVIEW

**Comment by jadenPete** [comment]
Hashing still presents an opportunity for non-determinism because two things can have the same hash, right?

File: src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala [hunk 1]
```diff
@@ -491,10 +491,45 @@ class AnxAnalyses(format: AnxAnalysisStore.Format) {
...
+      .clearClasspathHash()
+      .addAllClasspathHash(sortedClasspathHash)
+      .clearScalacOptions()
+      .addAllScalacOptions(sortedScalacOptions)
+      .clearJavacOptions()
+      .addAllJavacOptions(sortedJavacOptions)
+      .build()
+  }
+
+  def miniSetup(): Store[MiniSetup] = {
```
Comment (jroylance) [inline src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala:523]: I don't quite follow what's going on here - what is this function for?

Comment (jjudd) [reply src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala:523]: It's used to get a reader and writer for a MIniSetup. We have very similar functions for the other things on the proto schema. This modifies the writer to do the sorting when writing.

**Comment by jjudd** [comment]
> Hashing still presents an opportunity for non-determinism because two things can have the same hash, right?

I believe so, yes. Question is how often will we encounter that. And are the other things we could be sorting on.

**Review by jroylance — APPROVED** [review]

**Pull request merged**